### PR TITLE
Advanced batch mode

### PR DESCRIPTION
**Pull request by robinschmid**
I needed a way to run a batch on multiple datasets as a sequence easily.
Introduces an AdvancedParametersParameter that works with an accordion and checkbox to include optional advanced parameters. 
image

Will show a confirmation dialog so that people know they run in advanced mode
image

Imports one dataset after the other and creates a results folder with one folder for each dataset and the GNPS and SIRIUS outputs. 
image

### CODE REVIEW

File: src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupDialog.java [hunk 1]
```diff
@@ -52,8 +52,8 @@ public ParameterSetupDialog(boolean valueCheckRequired, ParameterSet parameters,
     this.paramsPane = createParameterPane(parameters.getParameters());
     centerPane.setCenter(paramsPane);
 
-    setMinWidth(500.0);
-    setMinHeight(400.0);
+    setMinWidth(200);
+    setMinHeight(200);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupDialog.java:56]: is this ok? maybe use setMinSize instead when setting both?

File: src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupPane.java [hunk 2]
```diff
@@ -109,6 +112,18 @@ public ParameterSetupPane(boolean valueCheckRequired, ParameterSet parameters,
...
+  }
+
+  /**
+   * Method to display setup dialog with a html-formatted footer message at the bottom.
+   *
+   * @param message: html-formatted text
+   */
+  public ParameterSetupPane(boolean valueCheckRequired, ParameterSet parameters,
+      boolean addOkButton, boolean addCancelButton, String message, boolean addParamComponents,
+      boolean addHelp) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupPane.java:126]: in which case would we not want the help?

File: src/main/java/io/github/mzmine/parameters/parametertypes/AdvancedParametersParameter.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import io.github.mzmine.parameters.UserParameter;
+import java.util.Collection;
+import java.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.w3c.dom.Element;
+
+/**
+ * @author Robin Schmid <a href="https://github.com/robinschmid">https://github.com/robinschmid</a>
+ */
+public class AdvancedParametersParameter<T extends ParameterSet> implements
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/parameters/parametertypes/AdvancedParametersParameter.java:33]: More javadoc to explain what this does and describe a usecase

File: src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupDialog.java (hunk 1 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupDialog.java:56]: there is no minSize function

File: src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java [hunk 4]
```diff
@@ -79,22 +102,125 @@ public void run() {
...
+    int currentDataset = -1;
+    String datasetName = "";
     // Process individual batch steps
     for (int i = 0; i < totalSteps; i++) {
+      // at the end of one dataset, clear the project and start over again
+      if (processedSteps % stepsPerDataset == 0) {
+        // clear the old project
+        MZmineCore.getProjectManager().clearProject();
+        currentDataset++;
+        logger.info(String.format("Starting batch dataset %d/%d", processedSteps / stepsPerDataset,
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java:124]: what do you mean

File: src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java [hunk 5]
```diff
@@ -79,22 +102,125 @@ public void run() {
...
+  }
+
+  private void setOutputFiles(final File parentDir, final boolean createResultsDir,
+      final String datasetName) {
+    int changedOutputSteps = 0;
+    for (MZmineProcessingStep<?> currentStep : queue) {
+      ParameterSet params = currentStep.getParameterSet();
+      // check some
+      for (final Parameter<?> p : params.getParameters()) {
+        if (p instanceof FileNameParameter fnp) {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java:185]: I guess we have to have a list somewhere of all export modules?

File: src/main/java/io/github/mzmine/modules/tools/clear_project/ClearProjectModule.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+public class ClearProjectModule implements MZmineProcessingModule {
+
+  @Override
+  public @NotNull String getName() {
+    return "Clear project";
+  }
+
+  @Override
+  public @Nullable Class<? extends ParameterSet> getParameterSetClass() {
+    return null;
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/tools/clear_project/ClearProjectModule.java:41]: maybe SimpleParameterSet.class will work? needs to be a class and empty

File: src/main/java/io/github/mzmine/modules/tools/clear_project/ClearProjectParameters.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with MZmine; if not,
+ * write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+package io.github.mzmine.modules.tools.clear_project;
+
+public class ClearProjectParameters {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/tools/clear_project/ClearProjectParameters.java:21]: that was actually empty :D
Removed it there should not be a random empty class.

File: src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupPane.java (hunk 2 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/parameters/dialogs/ParameterSetupPane.java:126]: At one point we can enforce it... but I guess there will be undocumented  small things

File: src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java:124]: It should say: "Processing dataset Lab_1 (1/15)" or somethig like that.

File: src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java:185]: yeah i think so

[the remark this answers is not in the document]
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java:185]: or tag the parameter in the export modules with an enum to let the batch mode know

[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchTask.java:192]: I am now checking if the module is defined as RawDataExport or FeatureListExport. I guess that should work best right?

File: src/main/java/io/github/mzmine/parameters/parametertypes/AdvancedParametersParameter.java (hunk 3 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/parameters/parametertypes/AdvancedParametersParameter.java:36]: Done

**Comment by robinschmid** [comment]
Addressed all comments hopefully.

**Pull request merged**